### PR TITLE
[dev] Reflect channel change in URL

### PR DESCRIPTION
**Pull request by Fab1en**
It means that if I go to a channel and refresh the page, I will be
reconnected to this channel after refresh.

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -784,6 +784,11 @@ class GlobalBindings {
     this.requestMove = (user, channel) => {
       if (this.connected()) {
         user.model.setChannel(channel.model)
+        // reflect this change in URL
+        let currentUrl = url.parse(document.location.href, true)
+        currentUrl.query.channelName = channel.name()
+        delete currentUrl.search
```
Comment (Johni0702) [inline app/index.js:790]: This is probably mostly my ignorance of the respective API but why is this necessary?

File: app/index.js [hunk 2]
```diff
@@ -784,6 +784,11 @@ class GlobalBindings {
     this.requestMove = (user, channel) => {
       if (this.connected()) {
         user.model.setChannel(channel.model)
+        // reflect this change in URL
+        let currentUrl = url.parse(document.location.href, true)
+        currentUrl.query.channelName = channel.name()
```
Comment (Johni0702) [inline app/index.js:789]: Does this work properly with nested channels?

File: app/index.js (hunk 1 above)
Comment (Fab1en) [reply app/index.js:790]: because `search` and `query` parameters point both to the same data (the string after `?`), but `search` is only the full string, whereas `query` is an object. As stated [in the doc](https://github.com/defunctzombie/node-url#urlformaturlobj), if both are provided only `search` is used.

File: app/index.js (hunk 2 above)
Comment (Fab1en) [reply app/index.js:789]: Good point, I haven't tested it and that probably does not work. Let me fix that.

**Comment by Fab1en** [comment]
I have made a fix to take sub-channels into account.

File: app/index.js [hunk 3]
```diff
@@ -784,6 +784,20 @@ class GlobalBindings {
     this.requestMove = (user, channel) => {
       if (this.connected()) {
         user.model.setChannel(channel.model)
+
+        // get full channel path
+        let path = channel.name()
+        let parent = channel.parent()
+        while( parent.name && parent.name() != 'Root' ){
```
Comment (Johni0702) [inline app/index.js:791]: If the user is already in the root channel, wouldn't `parent` be null? In that case we should probably skip adding the `channelName` alltogether.
Also, I don't think the name of the root channel is guaranteed to be `'Root'` (but its parent should always be null).

Comment (Fab1en) [reply app/index.js:791]: ### That's right, I get `TypeError: parent is undefined` if I go back to Root. Let's fix that.

Comment (Fab1en) [reply app/index.js]: I fixed it.
